ui/selectors/multichain-accounts/account-tree.ts

```diff
@@ -733,3 +719,92 @@ export const getAccountGroupsByAddress = createDeepEqualSelector(
     return [...matchingGroups];
   },
 );
+
+/**
+ * Selector to get a list of internal accounts spread across different network scopes for a specific account group.
+ *
+ * @param _state - Redux state.
+ * @param groupId - The ID of the account group.
+ * @returns An array of internal accounts spread across different network scopes.
+ */
+export const getInternalAccountListSpreadByScopesByGroupId =
+  createDeepEqualSelector(
+    [
+      getInternalAccountsFromGroupById,
+      getMultichainNetworkConfigurationsByChainId,
+    ],
+    (
+      internalAccounts: InternalAccount[],
+      networks: [
+        Record<CaipChainId, MultichainNetworkConfiguration>,
+        Record<Hex, NetworkConfiguration>,
+      ],
+    ): {
+      account: InternalAccount;
+      scope: CaipChainId;
+      networkName: string;
+    }[] => {
+      const caipNetworks = networks[0];
+
+      // Precompute EVM network IDs (filtered by non-test networks)
+      const evmNetworkIds = new Set(
+        Object.keys(caipNetworks).filter(
+          (chainId) =>
+            chainId.startsWith(KnownCaipNamespace.Eip155) &&
+            !isTestNetwork(chainId as CaipChainId),
+        ) as CaipChainId[],
```

Comment:
Why do we need to cast here?

Reply:
The filter is return type is a string array so we need to type cast it to match the expected data type of the consumers